### PR TITLE
sof: Add small enhancements

### PR DESCRIPTION
**Pull request by pnx9**
Modifications:
 -> Remove comparison to NULL

Signed-off-by: Sam Muhammed <jane.pnx9@gmail.com>

### CODE REVIEW

**Comment by dbaluta** [comment]
See error reported by checkpatch:

```
ERROR: spaces required around that '||' (ctx:VxW)
#87: FILE: test/cmocka/src/list/list_item_append.c:33:
+	if (!data->heads|| !data->tail_minus_1
```

**Comment by paulstelian97** [comment]
> See error reported by checkpatch:
> 
> ```
> ERROR: spaces required around that '||' (ctx:VxW)
> #87: FILE: test/cmocka/src/list/list_item_append.c:33:
> +	if (!data->heads|| !data->tail_minus_1
> ```

I'm not sure why it compiles with "heads" even. I think it doesn't (SOFCI UT are all failing)

**Comment by pnx9** [comment]
@dbaluta @paulstelian97 Oh, Sorry about that, i'll set up a post commit checkpatch.pl for these mistakes.

I guess building the SOF binaries after the changes won't tell me much??
any other tests to do on my target machine??

i usually run:  _ with docker_ for _byt_ platform
```
./scripts/xtensa-build-all.sh
```

but iam assuming it's not enough? it compiled with this error.

**Comment by paulstelian97** [comment]
> @dbaluta @paulstelian97 Oh, Sorry about that, i'll set up a post commit checkpatch.pl for these mistakes.
> 
> I guess building the SOF binaries after the changes won't tell me much??
> any other tests to do on my target machine??
> 
> i usually run: _ with docker_ for _byt_ platform
> 
> ```
> ./scripts/xtensa-build-all.sh
> ```
> 
> but iam assuming it's not enough? it compiled with this error.

I took the errors from what CI itself does. Even for my own pull requests I send them, see if CI complains (any error), check if my own platform builds (I don't use docker build and I don't know how well it works) and if it's all green I'll just wait for the reviews and merge to happen.

**Comment by pnx9** [comment]
> @pnx9 I will be taking some of this code into Zephyr (open source RTOS) which is Apache licence. I assume this is OK by you ?

Hi, I only do clean-up patches for the moment, this PR is for changes to code written by yourself and Janusz Jankowski. so it's absolutely OK.. Thank You ヽ(´▽`)/

**Comment by pnx9** [comment]
> > @dbaluta @paulstelian97 Oh, Sorry about that, i'll set up a post commit checkpatch.pl for these mistakes.
> > I guess building the SOF binaries after the changes won't tell me much??
> > any other tests to do on my target machine??
> > i usually run: _ with docker_ for _byt_ platform
> > ```
> > ./scripts/xtensa-build-all.sh
> > ```
> > 
> > 
> > but iam assuming it's not enough? it compiled with this error.
> 
> I took the errors from what CI itself does. Even for my own pull requests I send them, see if CI complains (any error), check if my own platform builds (I don't use docker build and I don't know how well it works) and if it's all green I'll just wait for the reviews and merge to happen.

Fair enough, CI does a qemu-check which i haven't tried yet. i guess i'll tinker with all these options -adding a native build as well- for the moment to get to the testing habit and see how it goes.
Thank you for pointing that out!  :)

**Comment by pnx9** [comment]
@jajanusz Hello

static int setup() and 
static int teardown()
are duplicated across list_item_is_last.c, list_item_prepend.c, list_item_del.c, list_item_append.c .

should both be refactored?
defined once and declared where needed, but where to store the definition, i guess <sof/list.h>?

I can work on this issue if it's needed.
please let me know, thank you.

**Comment by pnx9** [comment]
Some files has serious coding style issues, including lines of modified code, these were there already, i guess all of these coding style issues could be cleaned on a separete commit.

**Comment by pnx9** [comment]
> Do NOT modify src/arch/xtensa -- it's imported code from XCC. It cannot use <sof/bit.h> naturally.

Oops, sorry about that, I'll drop changes to xtensa

File: src/arch/xtensa/hal/mpu.c [hunk 1]
```diff
@@ -20,6 +20,8 @@
  * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
Comment (dbaluta) [inline src/arch/xtensa/hal/mpu.c:21]: @pnx9 sources inside src/arch/xtensa are coming for a 3rd party and we don't modify them. Please remove the patches touching this files.

Also, make sure you use BIT(x) when we really have to deal with a bit of a hardware register. In other cases, this macro can be misleading.

**Comment by dbaluta** [comment]
SOFCI TEST

**Comment by jajanusz** [comment]
> @jajanusz Hello
> 
> static int setup() and
> static int teardown()
> are duplicated across list_item_is_last.c, list_item_prepend.c, list_item_del.c, list_item_append.c .
> 
> should both be refactored?
> defined once and declared where needed, but where to store the definition, i guess <sof/list.h>?
> 
> I can work on this issue if it's needed.
> please let me know, thank you.

Heh, these were the first tests added to the repo, so yea they may be terrible.
You can refactor duplicated code, but it should NOT go to `sof/list.h` as it is header for project-wide stuff, I'd make local/private header in the same folder in which are unit tests as it is going to be used only by them, and you can include them with `""` instead of '<>'.
Thanks

**Pull request merged**